src/client/components/home/MealList.js

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useEffect, useState } from "react";
+import Meal from "./Meal";
+import "./meals.css";
+export const MealContext = createContext();
+function MealList() {
+  const [meals, setMeals] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    getResult();
+  }, []);
+  const getResult = async () => {
+    setLoading(true);
+    try {
+      const response = await fetch("/api/meals");
+      const result = await response.json();
+      console.log(result);
+      setMeals(result);
+    } catch (err) {
+      setError(err);
+    }
+    setLoading(false);
+  };
```

Comment:
```suggestion
  useEffect(() => {
    const getResult = async () => {
    setLoading(true);
    try {
      const response = await fetch("/api/meals");
      const result = await response.json();
      console.log(result);
      setMeals(result);
    } catch (err) {
      setError(err);
    }
    setLoading(false);
  };
  
    getResult();
  }, []);
```

It's more common to see this pattern this way, where you first define the async function inside the effect and then invoke it. Avoids issues with hoisting and makes it more clear what the effect does since the actual code is first.